manifests/candlepin.pp

```diff
@@ -132,27 +128,33 @@
       key_owner   => $user,
       key_group   => $client_keypair_group,
       key_mode    => '0440',
-    } ~>
+    }
+
     file { $truststore_password_path:
       ensure  => file,
       content => $truststore_password,
       owner   => 'root',
       group   => $group,
       mode    => '0440',
-    } ~>
-    exec { 'Create Candlepin truststore with CA':
-      command => "keytool -import -v -keystore ${truststore} -alias ${alias} -file ${ca_cert} -noprompt -storetype pkcs12 -storepass:file ${truststore_password_path}",
-      unless  => "keytool -list -keystore ${truststore} -alias ${alias} -storepass:file ${truststore_password_path}",
-    } ~>
+    }
+
+    truststore_certificate { "${truststore}:${alias}":
+      ensure        => present,
+      password_file => $truststore_password_path,
+      certificate   => $ca_cert,
+    }
+
+    truststore_certificate { "${truststore}:${artemis_alias}":
+      ensure        => present,
+      password_file => $truststore_password_path,
+      certificate   => $client_cert,
+    }
+
     file { $truststore:
```

Comment:
What prevents this file resource from executing before the truststore_certificate call? Can that lead to any file type or permission mismatches if the file is ensured before one of the truststore_certificate resources are called? 

Is this a hidden relationship because of the `autonotify(:file)` block?